route-rs-runtime/src/io/pcap.rs

```diff
@@ -0,0 +1,212 @@
+                let cap_start = if let Some(p) = cap_iter.peek() {
+                    p.timestamp
+                } else {
+                    Duration::from_secs(0)
+                };
+                PcapReplayState::RunRealtime(FromPcapRealtimeState { start, cap_start })
+            }
+        };
+        FromPcapProcessor {
+            src_file,
+            capture: cap_iter,
+            replay_state,
+        }
+    }
+}
+
+struct PcapPacketWrap {
+    timestamp: Duration,
+    data: Vec<u8>,
+}
+// Wrapper to pcap::Capture to create iterator
+// because pcap::Packet is basically just references to file data
+struct PcapCaptureWrap {
+    capture: Capture<Offline>,
+}
+impl Iterator for PcapCaptureWrap {
+    type Item = PcapPacketWrap;
+    fn next(&mut self) -> Option<Self::Item> {
+        if let Ok(p) = self.capture.next() {
+            let ts = p.header.ts;
+            //Why are time_t and suseconds_t defined as signed integers?
+            let timestamp =
+                Duration::from_micros(ts.tv_sec as u64) + Duration::from_secs(ts.tv_sec as u64);
+            let data: Vec<u8> = p.data.into();
+            Some(PcapPacketWrap { timestamp, data })
+        } else {
+            None
+        }
+    }
+}
+use tokio::prelude::task;
+
+impl Stream for FromPcapProcessor {
+    type Item = Vec<u8>;
+    type Error = ();
+
+    fn poll(&mut self) -> Poll<Option<Self::Item>, Self::Error> {
+        match &self.replay_state {
+            PcapReplayState::RunDump => {
+                if let Some(p) = self.capture.next() {
+                    let output_packet: Vec<u8> = p.data;
+                    Ok(Async::Ready(Some(output_packet)))
+                } else {
+                    Ok(Async::Ready(None))
+                }
+            }
+            PcapReplayState::RunRealtime(state) => {
+                let peek_intime = if let Some(p) = self.capture.peek() {
+                    (state.start.elapsed() > (p.timestamp - state.cap_start))
+                } else {
+                    //End of Pcap File
+                    return Ok(Async::Ready(None));
+                };
+                if peek_intime {
+                    if let Some(p) = self.capture.next() {
+                        let output_packet: Vec<u8> = p.data;
+                        println!(
+                            "Sending Time {:?} {:?} {:?}",
+                            state.start.elapsed(),
+                            p.timestamp,
+                            p.timestamp - state.cap_start
+                        );
+                        Ok(Async::Ready(Some(output_packet)))
+                    } else {
+                        unreachable!() // This is unexpected as logically peek believes a packet exists
+                    }
+                } else {
+                    // No packet to send reschedule
+                    // TODO: Should we rely on Delay instead and let that notify?
+                    task::current().notify();
```

Comment:
Since I anticipate that replaying pcaps is really only something done for testing purposes. I think this current notify loop is fine. Similar to Sam's stream->channel interface.